libevmasm/PeepholeOptimiser.cpp

```diff
@@ -57,6 +57,31 @@ struct PushPop
 	}
 };
 
+struct AddPop
+{
+	static size_t windowSize() { return 2; }
+	static bool apply(AssemblyItems::const_iterator _in, std::back_insert_iterator<AssemblyItems> _out)
+	{
+		if (_in[1] == Instruction::POP &&
+			_in[0].type() == Operation
+		)
+		{
+			Instruction i0 = _in[0].instruction();
+			if (instructionInfo(i0).ret == 1 &&
+				instructionInfo(i0).args == 2 &&
```

Comment:
We don't have to limit this to 2 arguments - we might even combine this with `PushPop`.